borsh/src/de/mod.rs

```diff
@@ -310,6 +290,38 @@ where
     }
 }
 
+impl<T> BorshDeserialize for VecDeque<T>
+where
+    T: BorshDeserialize,
+{
+    #[inline]
+    fn deserialize(buf: &mut &[u8]) -> Result<Self> {
+        let vec = <Vec<T>>::deserialize(buf)?;
+        Ok(vec.into())
+    }
+}
+
+impl<T> BorshDeserialize for LinkedList<T>
+where
+    T: BorshDeserialize,
+{
+    #[inline]
+    fn deserialize(buf: &mut &[u8]) -> Result<Self> {
+        let vec = <Vec<T>>::deserialize(buf)?;
+        Ok(vec.into_iter().collect::<LinkedList<T>>())
```

Comment:
This code looks beautiful, however do we concern about avoid extra copy and convert cost, if we really want highest performance?

Reply:
That is how HashSet was already implemented. I think we can have another round of writing a helper to deserialize into an iterator.